contracts/Governance.sol

```diff
@@ -211,23 +222,160 @@ contract Governance is Context, Ownable, ERC165, EIP712 {
         return _votingPeriod;
     }
 
-    function setTreasury(address treasury_) external virtual onlyOwner {
+    function setInitialTreasury(address treasury_) external virtual onlyOwner {
```

Comment:
suggestion: have this initial setting of the treasury happen during the constructor (as the owner sets it).

Seem strange to have two very similar methods with the main difference being the modifier `onlyOwner` or `onlyGovernance`. Is the setting of the Treasury sit with either the Owner or the Governance.

Reply:
Problem here is that I will need to deploy treasury first then before Governance, which doesn't make sense as Treasury needs governance for init too. Ofc, its possible, I can deploy Treasury first with governance and executor addresses as owner, then deploy Governance, passing treasury (which is kinda bad, because this means Governance always relies on treasury address and not backwards) and then init it.

Reply:
Maybe name it initTreasury then, same as you suggested below?

Reply:
Can't leave reply to below comment for some reason, but in general I don't want to have constructor overblown with too many inits, args, some of which even require loops and more special logic.